src/physics_swe.c

```diff
@@ -115,25 +115,23 @@ static PetscErrorCode RDyCeedOperatorSetUp(RDy rdy, PetscReal dt) {
   if (rdy->ceed_resource[0] && !rdy->ceed_rhs.op_edges) {
     rdy->ceed_rhs.dt = 0.0;
 
-    Ceed ceed;
-    CeedInit(rdy->ceed_resource, &ceed);
-    CeedCompositeOperatorCreate(ceed, &rdy->ceed_rhs.op_edges);
+    CeedCompositeOperatorCreate(rdy->ceed, &rdy->ceed_rhs.op_edges);
```

Comment:
Ideally, `RDyCeedOperatorSetUp()` shouldn't be called within a time step and done during initialization. Fixing this would require moving the call to `RDyCeedOperatorSetUp` outside of `RDyCeedOperatorApply()`. Should this be fixed in this PR itself or in another PR?

Reply:
I can move `RDyCeedOperatorSetup` outside of `RDyCeedOperatorApply` and see what happens.